pyrobosim/pyrobosim/navigation/execution.py

```diff
@@ -205,6 +226,75 @@ def validate_remaining_path(self) -> None:
 
             time.sleep(max(0, self.validation_dt - (time.time() - start_time)))
 
+    def detect_closed_hallway(self) -> None:
+        """
+        Get lidar measurement and determine if it's scanning a hallway.
+        If yes, it would update the robot recorded_closed_hallways knowledge.
+        It either remove (if it detects hallway is open) or add (if it detects hallway is close) the hallway
+        into the robot's knowledge.
+        """
+        if (self.robot is None) or (self.traj is None) or (self.robot.world is None):
+            return
+
+        while self.following_path and (not self.abort_execution):
+            start_time = time.time()
+            cur_pose = self.robot.get_pose()
+
+            lidar_sensor = self.robot.sensors.get("lidar")
```

Comment:
I feel this check should only happen once at the beginning, and should error if the robot does not have a lidar.

Also, it's not necessary that the lidar sensor is named `lidar`. This should not be hard-coded, but rather an input argument to the executor's constructor.